### PR TITLE
PMM-6038 PT summary.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**

### CODE REVIEW

**Review by BupycHuk — APPROVED** [review]

**Review by AlekSi — CHANGES_REQUESTED** [review]
Please never update api/api.descriptor. It should be updated only after each PMM release and only by @BupycHuk.

File: api/agentpb/agent.proto [hunk 1]
```diff
@@ -120,6 +120,9 @@ message QueryActionResult {
 
 // StartActionRequest is a ServerMessage asking pmm-agent to start action.
 message StartActionRequest {
+  // PTSummaryParams describes parameters for PT summary.
+  message PTSummaryParams {}
```
Comment (AlekSi) [inline api/agentpb/agent.proto:124]: Please move it after MongoDBExplainParams so the order is the same

**Comment by BupycHuk** [comment]
@AlekSi thanks, missed that.

File: api/managementpb/actions.proto [hunk 2]
```diff
@@ -17,6 +17,7 @@ enum ActionType {
   POSTGRESQL_SHOW_CREATE_TABLE = 5;
   POSTGRESQL_SHOW_INDEX = 6;
   MONGODB_EXPLAIN = 7;
+  PT_SUMMARY = 9;
```
Comment (AlekSi) [inline api/managementpb/actions.proto:20]: Why 9 and not 8? :)

File: api/managementpb/actions.proto [hunk 3]
```diff
@@ -282,6 +283,20 @@ message CancelActionRequest {
   ];
 }
 
+message StartPTSummaryActionRequest {
```
Comment (AlekSi) [inline api/managementpb/actions.proto:286]: Why we put it between `CancelActionRequest` and `CancelActionResponse`?

File: api/managementpb/actions.proto (hunk 2 above)
Comment (JiriCtvrtka) [reply api/managementpb/actions.proto]: My fault, I added same number like for PT sumary params before.

File: api/agentpb/agent.proto (hunk 1 above)
Comment (JiriCtvrtka) [reply api/agentpb/agent.proto]: Order fixed.

File: api/managementpb/actions.proto (hunk 3 above)
Comment (JiriCtvrtka) [reply api/managementpb/actions.proto:277]: I didnt notice that. Thanks.

**Review by AlekSi — CHANGES_REQUESTED** [review]
image

https://github.com/percona/pmm/pull/564#pullrequestreview-473450315

**Comment by JiriCtvrtka** [comment]
Ok, next time I will not commit api.descriptor changes.

File: api/agentpb/agent.proto [hunk 4]
```diff
@@ -213,6 +216,7 @@ message StartActionRequest {
     PostgreSQLShowCreateTableParams postgresql_show_create_table_params = 6;
     PostgreSQLShowIndexParams postgresql_show_index_params = 7;
     MongoDBExplainParams mongodb_explain_params = 8;
+    PTSummaryParams pt_summary_params = 10;
```
Comment (BupycHuk) [inline api/agentpb/agent.proto:219]: 9?

File: Gopkg.lock [hunk 5]
```diff
@@ -744,42 +744,6 @@
   revision = "1f47ba4663831f2a9c28a62a7de3ff8bc45078f0"
   version = "v1.30.0"
 
-[[projects]]
-  digest = "1:71b8d81402ecab6f1b0196ea30684b6f6df20569b72302a750ce50a5e9a74727"
-  name = "google.golang.org/protobuf"
```
Comment (BupycHuk) [inline Gopkg.lock:749]: Do we really don't use protobuf?

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: api/agentpb/agent.proto (hunk 4 above)
Comment (JiriCtvrtka) [reply api/agentpb/agent.proto:219]: @BupycHuk When I merged it with latest master 9 was used. google.protobuf.Duration timeout = 9; few lines below.

Comment (BupycHuk) [reply api/agentpb/agent.proto:219]: Ok

File: Gopkg.lock (hunk 5 above)
Comment (JiriCtvrtka) [reply Gopkg.lock]: Reverted, it happened during deps.

Comment (BupycHuk) [reply Gopkg.lock]: It's weird that CI didn't check that.

**Review by BupycHuk — APPROVED** [review]

File: Gopkg.lock (hunk 5 above)
Comment (JiriCtvrtka) [reply Gopkg.lock]: Yes, it is. I dont even open Gopkg.lock change details, because it not failed.

**Comment by JiriCtvrtka** [comment]
FB: https://github.com/Percona-Lab/pmm-submodules/pull/1065

**Pull request merged**